src/components/NumericStepper/__tests__/NumericStepper.spec.js

```diff
@@ -0,0 +1,152 @@
+
+        // CASE: Increase by 2 (to equal to 2) and make sure that number of onChange calls equal to 1
+        wrapper.find('StepperButton').at(1).prop('onClick')();
+        wrapper.update();
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(onChangeMock).toBeCalledWith(2);
+        expect(wrapper.find('input').prop('value')).toEqual('2');
+
+        // CASE: Increase by 2 (to equal to 4) and make sure that number of onChange calls equal to 2
+        wrapper.find('StepperButton').at(1).prop('onClick')();
+        wrapper.update();
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(4);
+        expect(wrapper.find('input').prop('value')).toEqual('4');
+    });
+    it('should disable buttons when limits are reached', () => {
+        wrapper = mount(
+            <NumericStepper onChange={onChangeMock} minValue={1} maxValue={3} defaultValue={2} />
+        );
+
+        // Simulate two clicks on stepUp button
+        wrapper.find('StepperButton').at(1).prop('onClick')();
+        wrapper.update();
+
+        expect(wrapper.find('input').prop('value')).toEqual('3');
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(wrapper.find('StepperButton').at(1).prop('disabled')).toBe(true);
+
+        // Simulate two clicks on stepDown button
+        wrapper.find('StepperButton').at(0).prop('onClick')();
+        expect(onChangeMock).toBeCalledTimes(2);
+        wrapper.update();
+        wrapper.find('StepperButton').at(0).prop('onClick')();
+        expect(onChangeMock).toBeCalledTimes(3);
+        wrapper.update();
+
+        expect(wrapper.find('input').prop('value')).toEqual('1');
+        expect(wrapper.find('StepperButton').at(0).prop('disabled')).toBe(true);
+    });
+    it('should set bottom edge value if user enters value below allowed limit', () => {
+        wrapper = mount(<NumericStepper onChange={onChangeMock} minValue={10} />);
+        const data = { target: { value: '9' } };
+
+        wrapper.find('input').simulate('change', data);
+        wrapper.find('input').simulate('blur');
+
+        expect(wrapper.find('input').prop('value')).toEqual('10');
+    });
+    it('should set top edge value if user enters value above allowed limit', () => {
+        wrapper = mount(<NumericStepper onChange={onChangeMock} maxValue={10} />);
+        const data = { target: { value: '11' } };
+
+        wrapper.find('input').simulate('change', data);
+        wrapper.find('input').simulate('blur');
+
+        expect(wrapper.find('input').prop('value')).toEqual('10');
+    });
+    it('should react on keyup/keydown press', () => {
+        /**
+         * TODO: Test with E2E tests.
+         * As we actually validate data on blur, its quite challenging to chain together mouseEnter, keyDown and blur.
+         * I've tried multiple ways to do it, and don't want to spend way more time on that.
+         */
```

Comment:
I think it is important to test the result of the callback  - `onChangeMock.toHaveBeenCalledWith`

Reply:
Added this check where we test callback behavior.